presto-pinot-toolkit/src/test/java/com/facebook/presto/pinot/query/TestPinotQueryGenerator.java

```diff
@@ -374,4 +377,41 @@ public void testAggregationWithOrderByPushDownInTopN()
         TopNNode topN = new TopNNode(planBuilder.getIdAllocator().getNextId(), agg, 50L, new OrderingScheme(ImmutableList.of(new Ordering(variable("city"), SortOrder.DESC_NULLS_FIRST))), TopNNode.Step.FINAL);
         testPinotQuery(pinotConfig, topN, "", defaultSessionHolder, ImmutableMap.of());
     }
+
+    @Test
+    public void testDistinctLimitPushdown()
+    {
+        PlanBuilder planBuilder = createPlanBuilder(defaultSessionHolder);
+        DistinctLimitNode distinctLimitNode = distinctLimit(
+                planBuilder,
+                ImmutableList.of(new VariableReferenceExpression("regionid", BIGINT)),
+                50L,
+                tableScan(planBuilder, pinotTable, regionId));
+        testPinotQuery(
+                pinotConfig,
+                distinctLimitNode,
+                String.format("SELECT %s FROM realtimeOnly GROUP BY regionId %s 50", getExpectedDistinctOutput("count(*)", "regionId"), getGroupByLimitKey()),
+                defaultSessionHolder,
+                ImmutableMap.of());
+
+        planBuilder = createPlanBuilder(defaultSessionHolder);
+        distinctLimitNode = distinctLimit(
+                planBuilder,
+                ImmutableList.of(
+                        new VariableReferenceExpression("regionid", BIGINT),
+                        new VariableReferenceExpression("city", VARCHAR)),
+                50L,
+                tableScan(planBuilder, pinotTable, regionId, city));
+        testPinotQuery(
+                pinotConfig,
+                distinctLimitNode,
+                String.format("SELECT %s FROM realtimeOnly GROUP BY regionId, city %s 50", getExpectedDistinctOutput("count(*)", "regionId, city"), getGroupByLimitKey()),
+                defaultSessionHolder,
+                ImmutableMap.of());
+    }
+
+    protected String getExpectedDistinctOutput(String aggregation, String groupKeys)
+    {
+        return aggregation;
```

Comment:
I think getExpectedDistinctOutput should only take the group by keys, because the aggregation is always unconditionally "count(*)". ie, this method should just return "count(*)".

Reply:
make sense. Changed.